### DISCUSSION TITLE
How to get number of Intel GPUs without dpctl?

### DISCUSSION THREAD

**Discussion by harborn** [IntelPython/dpctl#1257] [General]
Hi,

I have a question about get device count of Intel GPUs.

By using dpctl, I can easily get count with ```len(dpctl.get_devices("level_zero", "gpu"))```
However, using this API should import dpctl,
If I don't want to import dpctl, how can I get the device count?
Like Nvidia GPU detection: by check the driver list in folder ```/proc/driver/nvidia/gpus```

Thanks.

**Comment by diptorupd** [comment]
Dpctl uses the dpcpp SYCL runtime, so the count of devices correlate to what the SYCL runtime exposes. For example, you can have a Nvidia GPU on your system, but it will not be counted as a device unless you specially build dpcpp and dpctl with Nvidia support (ref #1124).

If you do not want to use dpctl, then you can use `sycl-ls` to get the list of devices that are supported by your SYCL runtime. I do not recommend looking at driver directories as dpcpp can support different types of drivers: opencl, level_zero, cuda, rocm (last two need separate plugins).

**Comment by diptorupd** [comment]
Converting the issue to a discussion.

**Comment by oleksandr-pavlyk** [comment]
You can do 

```sh
$ python -m dpctl -s | wc -l
4
```

or

```sh
$ sycl-ls | wc -l
4
```
